pkg/registry/servicecatalog/broker/strategy_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+import (
+	"testing"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+// TestBrokerStrategyTrivial is the testing of the trivial hardcoded
+// boolean flags.
+func TestBrokerStrategyTrivial(t *testing.T) {
+	if createStrategy.NamespaceScoped() {
+		t.Errorf("broker create must not be namespace scoped")
+	}
+	if updateStrategy.NamespaceScoped() {
+		t.Errorf("broker update must not be namespace scoped")
+	}
+	if updateStrategy.AllowCreateOnUpdate() {
+		t.Errorf("broker should not allow create on update")
+	}
+	if updateStrategy.AllowUnconditionalUpdate() {
+		t.Errorf("broker should not allow unconditional update")
+	}
+}
+
+// TestBrokerCreate
+func TestBroker(t *testing.T) {
+	// Create a broker or brokers
+	broker := &sc.Broker{
+		Spec: sc.BrokerSpec{
+			URL:          "abcd",
+			AuthUsername: "user",
+			AuthPassword: "pass",
+			OSBGUID:      "guid",
+		},
+		Status: sc.BrokerStatus{
+			Conditions: nil,
+		},
+	}
+
+	// Canonicalize the broker
+	createStrategy.PrepareForCreate(nil, broker)
+
+	if broker.Status.Conditions == nil {
+		t.Fatalf("Fresh broker should have empty status")
+	}
+	if len(broker.Status.Conditions) != 0 {
+		t.Fatalf("Fresh broker should have empty status")
```

Comment:
@duglin I can't find the comment either, but regarding your `nil` vs `[]thing` question - the following 2 code snippets are semantically the same:

```go
var a []string = nil
```

and

```go
var a = []string{}
```

The difference is that the former doesn't eagerly allocate and the latter does. Often, this difference doesn't really matter because you'll be writing to the slice anyway, which will allocate.

FWIW, the most performant option is the following (requires that you know the length of the slice a priori):

```go
var a = make([]string, lengthOfSlice)
```